File: store_data_at_delete-setting-case-java/src/main/java/com/example/demo/memo/MemoEntity.java
package com.example.demo.memo;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.envers.Audited;

import java.time.LocalDateTime;

@Audited
@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MemoEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String title;
    @Column(nullable = false, columnDefinition = "MEDIUMTEXT")
    private String content;

    @Column(updatable = false)
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public MemoEntity cloneForUpdate(MemoEntity entity) {
        return new MemoEntity(id, entity.getTitle(), entity.getContent(), createdAt, LocalDateTime.now());
    }
}

File: store_data_at_delete-setting-case/src/main/kotlin/com/example/demo/memo/MemoEntity.kt
package com.example.demo.memo

import jakarta.persistence.*
import org.hibernate.envers.Audited
import java.time.LocalDateTime

@Audited
@Entity
data class MemoEntity(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long = 0L,

    @Column(nullable = false, length = 100)
    val title: String,
    @Column(nullable = false, columnDefinition = "MEDIUMTEXT")
    val content: String,

    @Column(updatable = false)
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime,
) {
    fun cloneForUpdate(editEntity: MemoEntity): MemoEntity {
        return this.copy(title = editEntity.title, content = editEntity.content, updatedAt = LocalDateTime.now())
    }
}
